src/Model/Contact.php

```diff
@@ -704,22 +704,31 @@ public static function updateSelfFromUserID(int $uid, bool $update_avatar = fals
 
 		$file_suffix = 'jpg';
 
-		$fields = ['name' => $profile['name'], 'nick' => $user['nickname'],
-			'avatar-date' => $self['avatar-date'], 'location' => Profile::formatLocation($profile),
-			'about' => $profile['about'], 'keywords' => $profile['pub_keywords'],
-			'contact-type' => $user['account-type'], 'prvkey' => $user['prvkey'],
-			'pubkey' => $user['pubkey'], 'xmpp' => $profile['xmpp'], 'matrix' => $profile['matrix'], 'network' => Protocol::DFRN];
+		$fields = [
+			'name'         => $profile['name'],
+			'nick'         => $user['nickname'],
+			'avatar-date'  => $self['avatar-date'],
+			'location'     => Profile::formatLocation($profile),
+			'about'        => $profile['about'],
+			'keywords'     => $profile['pub_keywords'],
+			'contact-type' => $user['account-type'],
+			'prvkey'       => $user['prvkey'],
+			'pubkey'       => $user['pubkey'],
+			'xmpp'         => $profile['xmpp'],
+			'matrix'       => $profile['matrix'],
+			'network'      => Protocol::DFRN,
+			'url'          => DI::baseUrl() . '/profile/' . $user['nickname'],
+			'addr'         => $user['nickname'] . '@' . substr(DI::baseUrl(), strpos(DI::baseUrl(), '://') + 3),
+			'request'      => DI::baseUrl() . '/dfrn_request/' . $user['nickname'],
+			'notify'       => DI::baseUrl() . '/dfrn_notify/' . $user['nickname'],
+			'poll'         => DI::baseUrl() . '/dfrn_poll/'. $user['nickname'],
+			'confirm'      => DI::baseUrl() . '/dfrn_confirm/' . $user['nickname'],
+			'poco'         => DI::baseUrl() . '/poco/' . $user['nickname'],
+		];
 
 		// it seems as if ported accounts can have wrong values, so we make sure that now everything is fine.
-		$fields['url'] = DI::baseUrl() . '/profile/' . $user['nickname'];
 		$fields['nurl'] = Strings::normaliseLink($fields['url']);
```

Comment:
Please extract the `$fields['url']` in a variable so you can include these two statements in the array declaration.